### ISSUE TITLE
[meta] drop cronet

### ISSUE THREAD

**Issue by Bnyro** [libre-tube/LibreTube#6867]
### Describe your suggested feature

F-Droid recently started marking Cronet unfree and thus LibreTube must be dropping that dependency to continue being updated on F-Droid, see https://github.com/libre-tube/LibreTube/issues/6455.

The whole downloader depends on Cronet, so that'll require a large refactor.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by FireMasterK** [comment]
https://gitlab.com/fdroid/fdroid-suss/-/blob/master/suss/com.google.android.gms.yml

It seems to just be the media3 cronet dependency. This is likely because of this:
https://github.com/androidx/media/blob/1a7da45dd9c6419a43fab95132ed0aa0999300aa/libraries/datasource_cronet/build.gradle#L27

Maybe we can ask them if it's okay to use it without the gms dependency?

**Comment by Bnyro** [comment]
I don't think that inlining the whole Cronet source code is a good idea and I don't want to maintain a fork of an external library either, so I'll probably remove Cronet with the next release until there's a better solution.
